pkg/client/cache/delta_fifo.go

```diff
@@ -404,6 +416,16 @@ func (f *DeltaFIFO) Pop(process PopProcessFunc) (interface{}, error) {
 	defer f.lock.Unlock()
 	for {
 		for len(f.queue) == 0 {
+			// When the queue is empty, invocation of Pop() is blocked until new item is enqueued.
```

Comment:
To make it clear: seeing existing data in the queue before the close is fine. But being able to keep the queue full and ignoring the close is probably not fine.